tvshows/src/components/features/review/StarRating.tsx

```diff
@@ -0,0 +1,66 @@
+import { Flex, Icon} from "@chakra-ui/react";
+import { StarIcon } from "@chakra-ui/icons";
+import { useState, useEffect } from "react";
+
+export interface IStarRating {
+  defaultValue: number;
+  onChange: (value: number) => void;
+  mode: number;
+}
+
+const StarRating = ({ defaultValue = 0, onChange, mode }: IStarRating) => {
+  const [rating, setRating] = useState(defaultValue);
+  const [hover, setHover] = useState<number | null>(null);
+
+  useEffect(() => {
+    setRating(defaultValue);
+  }, [defaultValue]);
+
+  const handleClick = (value: number) => {
+    setRating(value);
+    if (onChange) {
+      onChange(value);
+    }
+  };
+
+  const IconMode1 = (index: number, starValue: number) => {
+    return (
+      <Icon
```

Comment:
Not bad, but you could have achieved the same by just not passing some handlers to the same star component.